apis/v1alpha1/route_types.go

```diff
@@ -35,6 +35,23 @@ const (
 	GatewayAllowSameNamespace GatewayAllowType = "SameNamespace"
 )
 
+const (
+	// AnnotationAppProtocol defines the protocol a Gateway should use for
+	// communication with a Kubernetes Service. The annotation serves as a stop-gap
+	// solution for the Kubernetes versions < 1.18, where the `AppProtocol` field
+	// on the `Service` resource is not available. Using this annotation is highly
+	// discouraged when AppProtocol field is available. AppProtocol field, when
+	// populated, takes precedence over this annotation.
+	//
+	// The annotation value must be a comma separated with each value containing
+	// a port number followed by a forward slash and the protocol.
+	// Examples:
+	//
+	// - `networking.x-k8s.io/app-protocol: 8000/http,8443/https`
+	// - `networking.x-k8s.io/app-protocol: 4000/tcp,5000/tls`
```

Comment:
Is `/` used elsewhere for this kind of differentiation? If we used BackendPolicy here we could simply have a single annotation mapping to a protocol. Although BackendPolicy is not useful for much else right now, it will stay that way unless we start adding config there when it makes sense.

Reply:
The `AppProtocol` field uses `/` ...

> Un-prefixed names are reserved for IANA standard service names (as per RFC-6335 and http://www.iana.org/assignments/service-names). Non-standard protocols should use prefixed names such as mycompany.com/my-custom-protocol.

Reply:
Although AppProtocol uses `/` to differentiate, this is only tangentially related.
I initially went with `:` but then changed it to `/` to avoid parsing and reading issues in YAML.
As noted earlier, I don't have a preference here as long as we can encode the necessary data into a string.

> BackendPolicy is not useful for much else right now, it will stay that way unless we start adding config there when it makes sense.

BackendPolicy makes sense, especially for cases where the upstream resource is NOT a k8s Service.
We didn't have enough consensus (I might have read the feelings wrong) so I decided to write up more of a stop-gap solution.

Reply:
I think an annotation on BackendPolicy may be the right balance for now. It allows us to easily transition that annotation into a field if this ends up being widely used.